### PR TITLE
test: include listen path suffix tests

### PR DESCRIPTION
**Pull request by mvdan**
To be extra sure about issues like #1297.

### CODE REVIEW

File: gateway_test.go [hunk 1]
```diff
@@ -921,6 +921,9 @@ func TestListener(t *testing.T) {
 		{method: "GET", path: "/sample", code: 404},
 		{method: "GET", path: "/tyk/reload/?block=true", adminAuth: true, code: 200},
 		{method: "GET", path: "/sample", code: 200},
+		{method: "GET", path: "/samplefoo", code: 200},
+		{method: "GET", path: "/sample/", code: 200},
+		{method: "GET", path: "/sample/foo", code: 200},
```
Comment (dencoded) [inline gateway_test.go]: somewhat to mention, when it comes to standard string-values I would prefer to use already defined constants vs typing them (source of errors), like for http-verbs defined in `net/http` we have `MethodGet `

Comment (mvdan) [reply gateway_test.go]: What constant are you talking about? `"GET"`, `"/sample"`, or `200`?

All three of them are short, so I'm not sure that named constants are necessary. `net/http` itself is full of `"GET"` literals, for example. The constants there are defined for consistency more than for usefulness.

Comment (mvdan) [reply gateway_test.go]: Also worth noting that a typo in any of these literals would result in a test failure or panic, so I'm not too worried about the mistakes we could make either.

Comment (dencoded) [reply gateway_test.go]: no worries, the test code is totally fine. I meant that for http-verbs like `GET` (and yeah - response codes like `200`) we have predefined constants in `net/http` which we could use - `http.MethodGet` and `http.StatusOK`

**Pull request merged**